tests/test_weather.py

```diff
@@ -0,0 +1,76 @@
+    mocker.patch("src.weather.WeatherCall.geo_response", return_value = mock_json_return )
+    try:
+        #act
+        WeatherCall.get_coordinates('roma', 1)
+    except TypeError:
+        #assert
+        assert False
+    assert WeatherCall.get_coordinates('roma', 1) == mock_json_return
+
+def test_get_coordinates_fail(mocker: MockerFixture):
+    '''tests the get_coordinates function when the response gives a 401 code'''
+    #arrange
+    mock_json_return = {"cod":401, "message": "Invalid API key. Please see https://openweathermap.org/faq#error401 for more info."}
+    mocker.patch("src.weather.WeatherCall.geo_response", return_value = mock_json_return )
+    try:
+        #act
+        WeatherCall.get_coordinates('roma', 1)
+    except TypeError:
+        #assert
+        assert True
+
+def test_get_weather_passes(mocker: MockerFixture):
+    '''test the get_weather function when the response gives the correct json'''
+    #arrange
+    mock_json_return = {"coord":{"lon":12.4829,"lat":41.8933},"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01n"}],"base":"stations","main":{"temp":285.51,"feels_like":284.76,"temp_min":283.38,"temp_max":287.2,"pressure":1006,"humidity":75},"visibility":10000,"wind":{"speed":1.03,"deg":110},"clouds":{"all":0},"dt":1669221304,"sys":{"type":2,"id":2000926,"country":"IT","sunrise":1669183751,"sunset":1669218241},"timezone":3600,"id":3169070,"name":"Rome","cod":200}
+    mocker.patch("src.weather.WeatherCall.weather_response", return_value = mock_json_return )
+    try:
+        #act
+        WeatherCall.get_weather(41.8933203, 12.4829321)
```

Comment:
Perchè lo richiami qui e lo richiami sotto? Ti direi di sostiture la riga 60 con la riga con l'assert in basso.
O Altrimenti salvare la response e fare l'assert su di questa